src/python/WMCore/WMSpec/WMTask.py

```diff
@@ -926,10 +927,19 @@ def setSubscriptionInformation(self, custodialSites = None, nonCustodialSites =
             outputDataset = entry["outputDataset"]
             outputModule = entry["outputModule"]
 
-            primDs = outputDataset.split('/')[1]
-            tier = outputDataset.split('/')[3]
+            dsSplit = outputDataset.split('/')
+
+            primDs = dsSplit[1]
+
+            procDsSplit = dsSplit[2].split('-')
+            skim = ( len(procDsSplit) == 4 )
```

Comment:
But as useSkim has default False, you'll just use compatibility mode, no change.